lib/net.c

```diff
@@ -622,6 +622,22 @@ lua_init_iface(struct gatekeeper_if *iface, const char *iface_name,
 		else if (gk_type == AF_INET6 &&
 				inet_pton(AF_INET6, ip_addr,
 				&iface->ip6_addr) == 1) {
+			/*
+			 * According to RFC 6164, addresses with all zeros
+			 * in the rightmost 64 bits SHOULD NOT be assigned as
+			 * unicast addresses; addresses in which the rightmost
+			 * 64 bits are assigned the highest 128 values
+			 * (i.e., ffff:ffff:ffff:ff7f to ffff:ffff:ffff:ffff)
+			 * SHOULD NOT be used as unicast addresses.
+			 */
+			uint64_t *paddr = (uint64_t *)iface->ip6_addr.s6_addr;
+			uint64_t addr2 = be64toh(paddr[1]);
```

Comment:
Drop variable `paddr` and initialize addr2 with `rte_be_to_cpu_64(((rte_be64_t *)iface->ip6_addr.s6_addr)[1]);`